src/utils/roomUtil.ts

```diff
@@ -0,0 +1,46 @@
+import { ExampleRoom1Handler } from '../roomHandlers/exampleRoom1Handler';
+import { RoomHandler } from '../roomHandlers/roomHandler';
+import { DeveloperError } from './developerError';
+import {
+  ROOM_TYPES_TUPLE,
+  RoomType,
+  AllRoomHandlersByRoomType,
+  NarrowedRoomState,
+  RoomState,
+} from './types';
+
+const allRoomHandlersByRoomType: AllRoomHandlersByRoomType = {
+  exampleRoom1: new ExampleRoom1Handler(),
+};
+
+export const roomUtil = {
+  coerceRoomState: <TRoomType extends RoomType>(
+    roomState: NarrowedRoomState<TRoomType> | null,
+  ): RoomState => {
+    if (roomState !== null) {
+      return roomState;
+    }
+
+    const roomHandler = roomUtil.getRandomRoomHandler();
+    return roomHandler.createRoomState();
+  },
+
+  getRandomRoomHandler: (): RoomHandler<RoomType> => {
+    const randomIndex = Math.floor(Math.random() * ROOM_TYPES_TUPLE.length);
+    const randomRoomType = ROOM_TYPES_TUPLE[randomIndex];
+
+    if (randomRoomType === undefined) {
+      throw new DeveloperError('Unreachable');
```

Comment:
Maybe a better error message here could be `Unreachable roomHandler, might need to run "npm run compile:gameStateSchema" to rebuild`

Reply:
I like specifying what to do if it is reached. In this case it's actually because TypeScript can't trust that `Math.floor(Math.random() * ROOM_TYPES_TUPLE.length)` creates an index within the bounds of `ROOM_TYPES_TUPLE`.

Regardless, I updated the error message.